### ISSUE TITLE
Attendance Analytics not tracking "Current" and "Previous" data.

### ISSUE THREAD

**Issue by TiffanyBelcher** [SparkDevNetwork/Rock#4046]
### Description
When using Attendance Analytics to pull current data from a Sunday the information given is "No Data Found" (I usually do this on Sunday evenings to gather Kids numbers and Volunteers numbers of that given Sunday) . If I try and pull the data on a Monday and use "Previous" week - still get "no data found".

This had been working fine for us until the V10 update. Asked the question in the community and a few other people are experiencing the same issue.

Tested in the prealpha site and it was fine (but was told prealpha is running different version)

### Steps to Reproduce
Enter Attendance Analytics
Choose Checkin Type where data was recently captures (For us its Sunday Kids Checkin)
Choose "Current Week" (if using on a Sunday, or "Previous Week" if using on Monday or after.
Choose your Schedule, Campus and Area of data that you want to see.
Click Update  

**Expected behavior:**

Before the V10 update I was able to see a graph/chart of the areas with attendance numbers for that given Sunday.

**Actual behavior:**

Now with all the selected data requested it says "No Data Found"

### Versions
* **Rock Version:** v10.1
* **Client Culture Setting:** [Enter Client Culture here]

**Comment by waynepaulj** [comment]
I am experiencing the same issue with Attendance Analytics since updating to release 10 - 10.1.
No matter how I select the last Sunday's date, I am not able to see attendance for the current (last) Sunday.  In this case 12/29/2019.
I have tried current week, last week, last month, manually entering date ranges. The only way I am able to retrieve the last Sunday is to make the ending date at least one week after the desired date.

**Comment by jimdenton** [comment]
I'm also having the same issue. This also affects past data, such as looking at the first week of December, I only get data (using Date Range) if I make the date range two weeks or more. This began after updating to version 10 or 10.1. When using previous week, I only get data if I enter 2 or more weeks, but no data for 1 week.

**Comment by jbgarrett12** [comment]
This is happening to me as well. If I click over to Attendees instead of Chart, it will show the proper data.

**Comment by hbcpn** [comment]
We've also experienced this since upgrading to v10.1
Thanks for submitting this Tiffany!

**Comment by jardine98** [comment]
We just upgraded to v10.1 today and are now experiencing this as well. But I believe that giving analytics has a similar issue.

**Comment by OnlyByGrace** [comment]
Having the same issue on out 10.1 install.  The workaround of picking a date range that goes 1 week in advance of the date I'm looking for is working though.

**Comment by jardine98** [comment]
Thank you Luke! I will use that workaround.


Thank you,
Crista Jardine
Database Administrator
Goldcreek.org <http://goldcreek.org/>

<https://www.facebook.com/goldcreek/>
<https://twitter.com/goldcreek?lang=en>
<https://www.instagram.com/goldcreekchurch/>
*Download our app today! <http://get.theapp.co/b6cf>*

This message is intended only for the individual(s) named. If you are not
the intended recipient, you are hereby notified that disclosing, copying,
distributing or taking any actions in reliance on the contents of this
information is strictly prohibited. Please notify the sender immediately if
you have received this email by mistake.

**Comment by ethan-sparkdevnetwork** [comment]
@TiffanyBelcher @waynepaulj @jimdenton  @jbgarrett12 @OnlyByGrace @jardine98 
I am having a hard time reproducing this using test data. I am wondering if the SundayDate for the AttendanceOccurrence that is stored in the DB is incorrect. This query will try to find incorrect dates, can you run it using `Admin Tools --> Power Tools --> SQL Command`? Any rows returned would indicate a problem. If nothing is returned then the Sunday Dates are good.

```
SELECT g.[Name] as GroupName
	, ao.[OccurrenceDate]
	, ao.[SundayDate] as SundayDateInDB
	, dbo.ufnUtility_GetSundayDate(ao.[OccurrenceDate]) as CalculatedSundayDate
FROM [AttendanceOccurrence] ao
JOIN [Group] g ON ao.[GroupId] = g.[Id]
WHERE ao.[SundayDate] <> dbo.ufnUtility_GetSundayDate(ao.[OccurrenceDate])
```

**Comment by hbcpn** [comment]
@ethan-sparkdevnetwork just ran the SQL for Sunday dates with no results, so those should be good.

**Comment by ethan-sparkdevnetwork** [comment]
I just found the problem and fixed it in 10.2.

**Comment by ethan-sparkdevnetwork** [comment]
Fixed in commit: https://github.com/SparkDevNetwork/Rock/commit/5a3cbb703d7511bdaefa2dcd99dd9ebfa7f0afcf
